velox/connectors/hive/FileHandle.cpp

```diff
@@ -34,7 +34,7 @@ std::string groupName(const std::string& filename) {
 } // namespace
 
 std::shared_ptr<FileHandle> FileHandleGenerator::operator()(
-    const std::string& filename) {
+    const FileHandleKey& fileProperties) {
```

Comment:
You should keep path string as the key and set any extra properties (size, mtime, fd, etc.) after this call